compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -70,6 +70,10 @@ fn inline(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) -> bool {
     let mut this = Inliner {
         tcx,
         param_env: tcx.param_env_reveal_all_normalized(body.source.def_id()),
+        // FIXME: MIR optimization that run at mir-opt-level >= 2 might remove storage makers for
+        // some locals. This breaks compatibility with state transform that runs after inlining.
+        // As a workaround, we give such locals a storage for the duration of the call.
+        insert_storage_markers: tcx.sess.opts.debugging_opts.mir_opt_level >= 2,
```

Comment:
So the real fix would be to make sure we do not lose storage markers? Are you sure it's lost storage markers and not simply "never added" ones? I was sure we have lots of locals without storage markers. 

Or maybe we should just go for full https://github.com/rust-lang/rust/issues/68622 anyway...

Reply:
> So the real fix would be to make sure we do not lose storage markers?

This is what I had in mind originally, but we can also consider it to be a valid transformation even if it is suboptimal.

> I was sure we have lots of locals without storage markers.

Locals marked as internal do not participate in validation performed by the state transform. Though, In that context I think it makes sense to always limit storage of those locals simply to help stack space optimizations.

> maybe we should just go for full #68622 anyway...

It is unclear to me how the end of storage could be determined, especially for any local whose address escapes at some point.

Reply:
> It is unclear to me how the end of storage could be determined, especially for any local whose address escapes at some point.

Ah, we'd still have a "StorageDead" of sorts: `InvalidateBorrows`. It's not quite the same thing, but it will allow us to figure out that if we manually computed a `StorageDead`, it could not be before an `InvalidateBorrows` of the same local.